### PR TITLE
Added table for removed properties

### PR DESCRIPTION
**Pull request by mofterdinger**

### CODE REVIEW

File: java/migration.md [hunk 1]
```diff
@@ -55,26 +55,32 @@ Some property defaults have been adjusted:
...
-  - `cds.remote.remoteServiceConfig.destination.queries` -> `cds.remote.services.<key>.http.queries`
-  - `cds.sql.search.useLocalizedView` -> `cds.sql.search.model`
-  - `cds.identity.authConfig.enabled` -> `cds.security.authentication.authConfig.enabled`
-  - `cds.xsuaa.authConfig.enabled` -> `cds.security.authentication.authConfig.enabled`
-  - `cds.odataV4.apply.inCqn.enabled` -> `cds.odataV4.apply.transformations.enabled`
-  - `cds.multiTenancy.healthCheck.intervalMillis` -> `cds.multiTenancy.healthCheck.interval`
-  - `cds.messaging.services.<key>.outbox.persistent.enabled` -> `cds.messaging.services.<key>.outbox.name`
-  - `cds.auditlog.outbox.persistent.enabled` -> `cds.auditlog.outbox.name`
-  - `cds.outbox.persistent` -> `cds.outbox.services.<key>`
+Some deprecated properties are replaced with new properties. The following table gives an overview about the replaced properties:
```
Comment (beckermarc) [inline java/migration.md:58]: Maybe we shouldn't distignuish between replaced properties and removed properties. Basically all properties are removed. But some have an existing replacement, that is already available today. Currently it sounds like the replacement will only be introduced with 3.0.

File: java/migration.md [hunk 2]
```diff
@@ -55,26 +55,32 @@ Some property defaults have been adjusted:
...
+| `cds.xsuaa.authConfig.enabled` | `cds.security.authentication.authConfig.enabled` |
+| `cds.odataV4.apply.inCqn.enabled` | `cds.odataV4.apply.transformations.enabled` |
+| `cds.multiTenancy.healthCheck.intervalMillis` | `cds.multiTenancy.healthCheck.interval` |
+| `cds.messaging.services.<key>.outbox.persistent.enabled` | `cds.messaging.services.<key>.outbox.name` |
+| `cds.auditlog.outbox.persistent.enabled` | `cds.auditlog.outbox.name` |
+| `cds.outbox.persistent` | `cds.outbox.services.<key>` |
 
 ### Removed Properties
 
+Some deprecated properties are completely removed without a replacement:
```
Comment (beckermarc) [inline java/migration.md:82]: We should also explain WHY there are no replacements for these properties:

- supportedLocales: All locales are supported by default for localized entities, as session variables can now be leveraged on all databases
- unrestricted stuff: Maybe this is worth a dedicated chapter, as we also remove the methods there. We can link to the section in CAPire how to handle unrestricted attributes: https://pages.github.tools.sap/cap/docs/guides/security/authorization#unrestricted-xsuaa-attributes

File: java/migration.md (hunk 1 above)
Comment (mofterdinger) [reply java/migration.md]: ok, changed the text.

File: java/migration.md (hunk 2 above)
Comment (mofterdinger) [reply java/migration.md]: merged both props into the table

File: java/migration.md [hunk 3]
```diff
@@ -53,36 +53,31 @@ Some property defaults have been adjusted:
 | `cds.remote.services.<key>.http.csrf.enabled` | `true` | `false` | Most APIs don't require CSRF tokens. |
 
 
-### Replaced Properties
+### Changed Properties
```
Comment (beckermarc) [inline java/migration.md:56]: I think we can say they are removed.

File: java/migration.md [hunk 4]
```diff
@@ -53,36 +53,31 @@ Some property defaults have been adjusted:
...
-| `cds.auditlog.outbox.persistent.enabled` | `cds.auditlog.outbox.name` |
-| `cds.outbox.persistent` | `cds.outbox.services.<key>` |
-
-### Removed Properties
-
-Some deprecated properties are completely removed without a replacement:
-
-- `cds.sql.supportedLocales`
-- `cds.security.mock.users.<key>.unrestricted`
+| old Property | new Property | Explaination |
```
Comment (beckermarc) [inline java/migration.md:60]:
```suggestion
| Removed Property | Replacement | Explanation |
```

File: java/migration.md [hunk 5]
```diff
@@ -53,36 +53,31 @@ Some property defaults have been adjusted:
...
+| `cds.sql.search.useLocalizedView` | `cds.sql.search.model` | |
+| `cds.identity.authConfig.enabled` | `cds.security.authentication.authConfig.enabled` | |
+| `cds.xsuaa.authConfig.enabled` | `cds.security.authentication.authConfig.enabled` | |
+| `cds.odataV4.apply.inCqn.enabled` | `cds.odataV4.apply.transformations.enabled` | |
+| `cds.multiTenancy.healthCheck.intervalMillis` | `cds.multiTenancy.healthCheck.interval` | |
+| `cds.messaging.services.<key>.outbox.persistent.enabled` | `cds.messaging.services.<key>.outbox.name` | |
+| `cds.auditlog.outbox.persistent.enabled` | `cds.auditlog.outbox.name` | |
+| `cds.outbox.persistent` | `cds.outbox.services.<key>` | |
+| `cds.sql.supportedLocales` | none | All locales are supported by default for localized entities, as session variables can now be leveraged on all databases |
+| `cds.security.mock.users.<key>.unrestricted` | none | to be filled |
```
Comment (beckermarc) [inline java/migration.md:80]:
```suggestion
| `cds.sql.supportedLocales` | | All locales are supported by default for localized entities, as session variables can now be leveraged on all databases |
| `cds.security.mock.users.<key>.unrestricted` | | Special handling of unrestricted attributes has been removed, in favor of [explicit modelling](../guides/security/authorization#unrestricted-xsuaa-attributes). |
```

File: java/migration.md [hunk 6]
```diff
@@ -66,33 +66,34 @@ Some property defaults have been adjusted:
...
+| `cds.multiTenancy.healthCheck.intervalMillis` | `cds.multiTenancy.healthCheck.interval` | |
+| `cds.messaging.services.<key>.outbox.persistent.enabled` | `cds.messaging.services.<key>.outbox.name` | |
+| `cds.auditlog.outbox.persistent.enabled` | `cds.auditlog.outbox.name` | |
+| `cds.outbox.persistent` | `cds.outbox.services.<key>` | |
+| `cds.sql.supportedLocales` | | All locales are supported by default for localized entities, as session variables can now be leveraged on all databases. |
+| `cds.security.mock.users.<key>.unrestricted` | | Special handling of unrestricted attributes has been removed, in favor of [explicit modelling](../guides/security/authorization#unrestricted-xsuaa-attributes). |
+| `cds.multiTenancy.mtxs.enabled` | - | MTXS is enabled [by default](#removed-mtx-classic-support). |
+| `cds.multiTenancy.compatibility.enabled` | - | MtSubscriptionService API [has been removed](#removed-mtx-classic-support) and compatibility mode is no longer available. |
+| `cds.multiTenancy.security.deploymentScope` | - | HTTP-based tenant upgrade endpoints [have been removed](#removed-mtx-classic-support). |
```
Comment (beckermarc) [inline java/migration.md:96]:
```suggestion
| `cds.multiTenancy.mtxs.enabled` | | MTXS is enabled [by default](#removed-mtx-classic-support). |
| `cds.multiTenancy.compatibility.enabled` | | MtSubscriptionService API [has been removed](#removed-mtx-classic-support) and compatibility mode is no longer available. |
| `cds.multiTenancy.security.deploymentScope` | | HTTP-based tenant upgrade endpoints [have been removed](#removed-mtx-classic-support). |
```

File: java/migration.md [hunk 7]
```diff
@@ -66,33 +66,34 @@ Some property defaults have been adjusted:
...
-- `cds.multiTenancy.compatibility.enabled`
-- `cds.multiTenancy.security.deploymentScope`
+The following table gives an overview about the removed properties:
+
+| Removed Property | Replacement | Explanation |
+| --- | --- | --- |
+| `cds.data-source.csvInitializationMode` | `cds.data-source.csv.initializationMode` | |
+| `cds.data-source.csvFileSuffix` | `cds.data-source.csv.fileSuffix` | |
+| `cds.data-source.csvPaths` | `cds.data-source.csv.paths` | |
+| `cds.data-source.csvSingleChangeset` | `cds.data-source.csv.singleChangeset` | |
```
Comment (beckermarc) [inline java/migration.md:78]:
```suggestion
| `cds.dataSource.csvInitializationMode` | `cds.dataSource.csv.initializationMode` | |
| `cds.dataSource.csvFileSuffix` | `cds.dataSource.csv.fileSuffix` | |
| `cds.dataSource.csvPaths` | `cds.dataSource.csv.paths` | |
| `cds.dataSource.csvSingleChangeset` | `cds.dataSource.csv.singleChangeset` | |
```

**Review by beckermarc — APPROVED** [review]

**Pull request merged**